lib/client/user/mediator-topics/has_session.js

```diff
@@ -0,0 +1,32 @@
+var CONSTANTS = require('../../../constants');
+
+/**
+ * Initialsing a subscriber for logging out a user
+ *
+ * @param {object} userEntityTopics
+ * @param {UserClient} userClient - The User Client
+ */
+module.exports = function hasSessionUser(userEntityTopics, userClient) {
+
+  /**
+   *
+   * Handling logging out the current user.
+   *
+   * @param {object} parameters
+   * @param {string/number} parameters.topicUid  - (Optional)  A unique ID to be used to publish completion / error topics.
+   * @returns {*}
+   */
+  return function handleHasSession(parameters) {
+    var self = this;
+    parameters = parameters || {};
+    var hasSessionDoneTopic = userEntityTopics.getTopic(CONSTANTS.TOPICS.HAS_SESSION, CONSTANTS.DONE_PREFIX, parameters.topicUid);
```

Comment:
Done this kind of refactoring on https://github.com/feedhenry-raincatcher/raincatcher-message/pull/15/commits/2f9ee7c19b7e8a05b860bad2ee348d3c3d3a9f34 , with some in-code comments, please take a look when you guys have the chance.